File: Boj.단계별/12_정렬/javascript/18870.js
const data = require('fs').readFileSync('/dev/stdin').toString().split('\n')[1].split(' ').map(Number);
const sorted_data = [...new Set(data)].reduce((arr,cur) => { arr.push(cur); return arr},[]).sort((a,b) => a-b);
let i = 0;
const dic = sorted_data.reduce((dic,cur) => {dic.set(cur,i); i++; return dic},new Map())
console.log(data.reduce((result,cur) => {result += dic.get(cur)+" "; return result},""));

File: Boj.단계별/12_정렬/python/18870.py
import sys
input = sys.stdin.readline

n = int(input())
data = list(map(int,input().split()))
sorted_data = sorted(set(data))
data_dic = {sorted_data[i] : i for i in range(len(sorted_data))}
for i in range(len(data)):
    data[i] = data_dic[data[i]]
print(" ".join(map(str,data)))